packages/core/src/App/Components/Layout/Header/menu-link.tsx

```diff
@@ -0,0 +1,174 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Icon, Text } from '@deriv/components';
+import { useIsRealAccountNeededForCashier } from '@deriv/hooks';
+import { isExternalLink, isMobile, routes, getStaticUrl } from '@deriv/shared';
+import { observer, useStore } from '@deriv/stores';
+import { localize } from '@deriv/translations';
+import { BinaryLink } from 'App/Components/Routes';
+
+type TMenuLink = {
+    data_testid: string;
+    icon: string;
+    is_active: boolean;
+    is_disabled: boolean;
+    is_hidden: boolean;
+    link_to: string;
+    onClickLink: () => void;
+    suffix_icon: string;
+    text: string;
+};
+
+const MenuLink = observer(
+    ({
+        data_testid,
+        icon = '',
+        is_active,
+        is_disabled,
+        is_hidden,
+        link_to = '',
+        onClickLink,
+        suffix_icon = '',
+        text,
+    }: Partial<TMenuLink>) => {
+        if (is_hidden) return null;
```

Comment:
just for clarification, wasn't there any error for this early return like react says: 

React Hook "React.useState" is called conditionally. React Hooks must be called in the exact same order in every component render.eslint[react-hooks/rules-of-hooks](https://reactjs.org/docs/hooks-rules.html)

Reply:
I'll move it to the place after useState. I accidentally missed this error while working on code smells issues